src/timeman.cpp

```diff
@@ -32,41 +32,38 @@ namespace {
 
   enum TimeType { OptimumTime, MaxTime };
 
-  const int MoveHorizon   = 50;   // Plan time management at most this many moves ahead
-  const double MaxRatio   = 7.09; // When in trouble, we can step over reserved time with this ratio
-  const double StealRatio = 0.35; // However we must not steal time from remaining moves over this ratio
-
-
-  // move_importance() is a skew-logistic function based on naive statistical
-  // analysis of "how many games are still undecided after n half-moves". Game
-  // is considered "undecided" as long as neither side has >275cp advantage.
-  // Data was extracted from the CCRL game database with some simple filtering criteria.
-
-  double move_importance(int ply) {
-
-    const double XScale = 7.64;
-    const double XShift = 58.4;
-    const double Skew   = 0.183;
-
-    return pow((1 + exp((ply - XShift) / XScale)), -Skew) + DBL_MIN; // Ensure non-zero
-  }
-
   template<TimeType T>
-  int remaining(int myTime, int movesToGo, int ply, int slowMover) {
-
-    const double TMaxRatio   = (T == OptimumTime ? 1 : MaxRatio);
-    const double TStealRatio = (T == OptimumTime ? 0 : StealRatio);
-
-    double moveImportance = (move_importance(ply) * slowMover) / 100;
-    double otherMovesImportance = 0;
-
-    for (int i = 1; i < movesToGo; ++i)
-        otherMovesImportance += move_importance(ply + 2 * i);
-
-    double ratio1 = (TMaxRatio * moveImportance) / (TMaxRatio * moveImportance + otherMovesImportance);
-    double ratio2 = (moveImportance + TStealRatio * otherMovesImportance) / (moveImportance + otherMovesImportance);
-
-    return int(myTime * std::min(ratio1, ratio2)); // Intel C++ asks for an explicit cast
+  int remaining(int myTime, int myInc, int moveOverhead, int movesToGo, int ply)
+  {
+    double TRatio, sd = 8.5;
+    int mn = (ply + 1) / 2; // current move number for any side
+
+    /// In movestogo case we distribute time according to quadratic function with the maximum around move 20 for 40 moves in y minutes case.
+ 
+    if (movesToGo)
+    {
+        TRatio = (T == OptimumTime ? 1.0 : 6.0) / movesToGo;
+        if (mn <= 40)
+            TRatio *= (1.1 - 0.001 * (mn-20) * (mn-20));
+        else
+            TRatio *= 1.5;
```

Comment:
I tried to optimize this for x=40 moves in y time case. It will work fine also for x<40, while some small modifications will be needed for x>>40, as already commented above. The problem is that this discontinuity exists in real games. With 40/y time control we are usually low on time at mn=40, and with plenty of time at mn=41, and most of games will be over until mn=65.